backend/src/main/scala/cromwell/backend/BackendActor.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.backend
+
+import akka.actor.{ActorRef, Actor, ActorLogging}
+import akka.event.LoggingReceive
+import cromwell.backend.BackendActor.{Validate, AfterAll, BeforeAll, CreateJobExecutor}
+import cromwell.backend.model.JobDescriptor
+
+object BackendActor {
+
+  // Commands
+  sealed trait BackendActorMessage
+  case object BeforeAll extends BackendActorMessage
+  case class CreateJobExecutor(jobDescriptor: JobDescriptor) extends BackendActorMessage
+  case object AfterAll extends BackendActorMessage
+  case object Validate extends BackendActorMessage
+
+  // Events
+  sealed trait BackendActorEvent
+  case object BeforeAllCompleted extends BackendActorEvent
+  case class BeforeAllFailed(throwable: Throwable) extends BackendActorEvent
+  case class JobExecutorCreated(backendJobExecutor: ActorRef) extends BackendActorEvent
+  case class JobExecutorCreationFailed(throwable: Throwable) extends BackendActorEvent
+  case object AfterAllCompleted extends BackendActorEvent
+  case class AfterAllFailed(throwable: Throwable) extends BackendActorEvent
+  case object ValidationCompleted extends BackendActorEvent
+  case class ValidationFailed(throwable: Throwable) extends BackendActorEvent
+
+}
+
+/**
+  * Defines basic structure and functionality to initialize and make use of a backend through an Akka actor system.
+  * Backend functions should be implemented by each custom backend.
+  */
+trait BackendActor extends Backend with Actor with ActorLogging {
```

Comment:
Since this is the difficult bit, but would be identical in each of the various backends, why not make this concrete, with a `BackendActor(backend: Backend)` constructor.

That way, the difficult actor bit can be done once, here. And for every Backend, the implementer can just write a bunch of blocking functions. All of the special corralling around that blocking can be done once in the general purpose `BackendActor`?